scripts/west_commands/blobs.py

```diff
@@ -121,26 +146,107 @@ def list(self, args):
+        returns the path of this cached blob. If the blob is not yet cached,
+        the blob is downloaded into the auto-cache directory and the path of
+        the freshly cached blob is returned.
+        """
+        cached_blob = self.get_cached_blob(blob, [auto_cache_dir])
+        if cached_blob:
+            return cached_blob
+        name = Path(blob['path']).name
+        sha256 = blob['sha256']
+        self.download_blob(blob, auto_cache_dir / f'{name}.{sha256}')
+        cached_blob = self.get_cached_blob(blob, [auto_cache_dir])
+        assert cached_blob, f'Blob {name} still not cached in auto-cache.'
+        return cached_blob
+
+    def get_cached_blob(self, blob, cache_dirs) -> Path | None:
+        """
+        Look for a cached blob in the provided cache directories.
+        A blob may be stored using either its original name or suffixed with
+        its SHA256 hash (e.g. "<name>.<sha256>").
+        Return the first matching path, or None if not found.
+        """
+        name = Path(blob['path']).name
+        sha256 = blob["sha256"]
+        candidate_names = [
+            f"{name}.{sha256}",  # suffixed version
+            name,  # original blob name
+        ]
+
+        for cache_dir in cache_dirs:
+            if not cache_dir.exists():
+                continue
+            for name in candidate_names:
+                candidate_path = cache_dir / name
+                if (
+                    zephyr_module.get_blob_status(candidate_path, sha256)
+                    == zephyr_module.BLOB_PRESENT
+                ):
+                    return candidate_path
+        return None
+
+    def download_blob(self, blob, path):
+        '''Download a blob from its url to a given path.'''
+        url = blob['url']
         scheme = urlparse(url).scheme
-        self.dbg(f'Fetching {path} with {scheme}')
+        self.dbg(f'Fetching blob from url {url} with {scheme} to path: {path}')
         import fetchers
-        fetcher = fetchers.get_fetcher_cls(scheme)
 
+        fetcher = fetchers.get_fetcher_cls(scheme)
         self.dbg(f'Found fetcher: {fetcher}')
         inst = fetcher()
         self.ensure_folder(path)
         inst.fetch(url, path)
 
+    def fetch_blob(self, args, blob):
+        """
+        Ensures that the specified blob is available at its path.
+        If caching is enabled and the blob exists in the cache, it is copied
+        from there. Otherwise, the blob is downloaded from its URL and placed
+        at the target path.
+        """
+        path = Path(blob['abspath'])
+
+        # collect existing cache dirs specified as args, otherwise from west config
+        cache_dirs = args.cache_dirs
+        auto_cache_dir = args.auto_cache
+        if self.has_config:
+            if not cache_dirs:
+                cache_dirs = self.config.get('blobs.cache-dirs')
+            if not auto_cache_dir:
+                auto_cache_dir = self.config.get('blobs.auto-cache')
```

Comment:
If the user passed an empty string, it should take precedence over the config IMO, so comparing to `None`.

```suggestion
            if cache_dirs is None:
                cache_dirs = self.config.get('blobs.cache-dirs')
            if auto_cache_dir is None:
                auto_cache_dir = self.config.get('blobs.auto-cache')
```